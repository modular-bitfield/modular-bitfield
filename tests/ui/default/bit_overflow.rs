use modular_bitfield::prelude::*;

#[bitfield]
pub struct BitOverflowTest {
    #[default = -1]    // Negative value should fail
    invalid: B8,
}

fn main() {}