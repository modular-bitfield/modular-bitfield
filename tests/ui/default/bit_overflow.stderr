error[E0600]: cannot apply unary operator `-` to type `u8`
 --> tests/ui/default/bit_overflow.rs:5:17
  |
5 |     #[default = -1]    // Negative value should fail
  |                 ^^ cannot apply unary operator `-`
  |
  = note: unsigned values cannot be negated
help: you may have meant the maximum value of `u8`
  |
5 -     #[default = -1]    // Negative value should fail
5 +     #[u8::MAX]    // Negative value should fail
  |
